### ISSUE TITLE
[issue]: A special requirement, Make a Questionaire, I want to Assistant Agent ask user question and User reply

### ISSUE THREAD

**Issue by johnson7788** [microsoft/autogen#2818]
### Describe the issue

Please give me some inspiration. thank you，big brother!

I want to Assistant Agent get a question from a tool, and ask the question to user, and user input the answer, then Assistant check user answer suitable, if suitable, get next question, if not, let user reply again. Finally, all questions done。Make a summaries for all user answer.

**Comment by LittleLittleCloud** [comment]
There's an AutoGen.Net example similar to your case, `form-assistant`.

Where it helps user to fill a form. If user provides any useful information, the form assitant will fill the form using that information. And for any remaining information that needs to be filled, the form assistant will ask user to fill it.

The workflow is like below. The gist is to have one agent update form and one agent generate questions on missing information

```mermaid
graph TD
    A[User Agent] -->|fill form| B[Form Agent]
    B -->|result of saving progress| C[Assistant Agent]
    C -->|generate prompt to ask more information,
    or termnate conversation if all infomration is collected| A
```

**Comment by johnson7788** [comment]
@LittleLittleCloud Yes, you are right, i find it from here https://github.com/microsoft/autogen/pull/1911, However, I don't know .NET programming, so I will try to implement it using Python.

**Comment by LittleLittleCloud** [comment]
Sounds good, let us know your progress, ping me anytime if you have any questions

**Comment by jackgerrits** [comment]
Closing as no reply
